flowzone.yml

```diff
@@ -3145,17 +3267,26 @@ jobs:
 
     <<: *customWorkingDirectory
 
+    env:
+      AWS_RETRY_MODE: adaptive
+      AWS_MAX_ATTEMPTS: 10
+      ATTEMPTS: 5
+      TIMEOUT: 3
+
     steps:
       - *checkoutVersionedSha
       - *configureAWSCredentials
       - *getAWSCallerIdentity
+      - *convenienceFunctions
 
       - name: Create templates bucket
         id: make_bucket
         run: |
-          bucket="$(aws s3api list-buckets | jq -r '.Buckets[] | select(.Name=="${{ vars.AWS_S3_BUCKET }}").Name' || true)"
+          source '${{ steps.functions.outputs.with_backoff }}'
+
+          bucket="$(with_backoff aws s3api list-buckets | jq -r '.Buckets[] | select(.Name=="${{ vars.AWS_S3_BUCKET }}").Name' || true)"
           if [[ -z "$bucket" ]]; then
-              result="$(aws s3 mb "s3://cfn-$(uuidgen)-${{ inputs.aws_region }}" --region '${{ inputs.aws_region }}')"
+              result="$(with_backoff aws s3 mb "s3://cfn-$(uuidgen)-${{ inputs.aws_region }}" --region '${{ inputs.aws_region }}')"
```

Comment:
What is so unreliable about these aws calls that we need a retry backoff all over the place?

I hated that part of the remote-builder tests, are we using these tools incorrectly that they always need multiple attempts?

Reply:
@klutchell AWS have rate limits, so if 10 requests arrive from the same IP within a short time, jobs fail.

Reply:
we should add a comment explaining that or someone will delete the retries, probably me when I forget why we added them

Reply:
Good read: https://sre.google/sre-book/addressing-cascading-failures/

Reply:
Yeah, I've seen that. We should try this action sometime:
https://github.com/marketplace/actions/retry-step

It doesn't have automatic random backoffs, but we could add our own randomization with `on_retry_command`.

Reply:
Either way, a comment would be nice to explain why this is needed here.